### PR TITLE
CRM-16189, added function to check if financial type has Deferred Inc…

### PR DESCRIPTION
**Pull request by pradpnayak**
…ome Account is relationship

---
- CRM-16189: Improve support for Accrual Method bookkeeping
  https://issues.civicrm.org/jira/browse/CRM-16189

### CODE REVIEW

**Comment by pradpnayak** [comment]
will commit unit test soon

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -246,4 +246,62 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
...
+   * @param $params array
+   *
+   * @param $contributionID Integer
+   *
+   * @param $form Object
+   *
+   * @return string
+   *
+   */
+  public static function checkForValidFinancialType($params, $contributionID = NULL, $form = NULL) {
```
Comment (JoeMurray) [inline CRM/Financial/BAO/FinancialAccount.php]: This is a misleading function name. How about checkFinancialTypeHasDeferred ?

**Comment by JoeMurray** [comment]
Are the checks on this really failing, @yashodha ?

**Comment by seamuslee001** [comment]
@JoeMurray exploring the full console log i see this error "PHP Fatal error:  Call to undefined method CRM_Contribute_PseudoConstant::checkContributeSettings() in /home/jenkins/buildkit/build/core-8565-2sqn7/sites/all/modules/civicrm/CRM/Financial/BAO/FinancialAccount.php on line 263"

**Comment by JoeMurray** [comment]
Can you help teach me how to find this? Clicking on the Details link then the Console Log link on the left I get to https://test.civicrm.org/job/CiviCRM-Core-PR/10151/console but don't find that error.

**Comment by seamuslee001** [comment]
@JoeMurray Once your there on that console up the top there will be a link to the "full log"  which leads you to https://test.civicrm.org/job/CiviCRM-Core-PR/10151/consoleFull and then you can search for PHP Fatal

**Comment by pradpnayak** [comment]
depends on https://github.com/civicrm/civicrm-core/pull/8562

**Comment by colemanw** [comment]
@pradpnayak needs rebase + the minor cleanup Joe suggested.

**Comment by pradpnayak** [comment]
Jenkins, retest this please.

**Comment by colemanw** [comment]
@pradpnayak I'm giving this "merge on pass" label - it does need rebasing.

**Comment by pradpnayak** [comment]
Rebased PR

**Comment by pradpnayak** [comment]
jenkins, retest this please

**Comment by pradpnayak** [comment]
jenkins, retest this please

**Comment by pradpnayak** [comment]
jenkins, retest this please

**Comment by colemanw** [comment]
jenkins, retest this please

**Comment by pradpnayak** [comment]
Its ready to merge now. :)

**Pull request merged**